zkevm-circuits/src/evm_circuit/execution.rs

```diff
@@ -755,6 +755,51 @@ impl<F: Field> ExecutionConfig<F> {
                 let mut offset = 0;
 
                 self.q_step_first.enable(&mut region, offset)?;
+                // handle empty block conditions
+                let is_empty_block = block.txs.is_empty();
+                if is_empty_block {
+                    // set 0 for rows which constraints refer to
+                    let end_row = if !exact && block.evm_circuit_pad_to != 0 {
+                        block.evm_circuit_pad_to
+                    } else {
+                        2
+                    };
+
+                    for column in iter::empty()
+                        .chain([
+                            self.q_used,
+                            self.q_step,
+                            self.num_rows_until_next_step,
+                            self.num_rows_inv,
+                            self.q_step_last,
+                        ])
+                        .chain(self.advices)
+                    {
+                        for i in 0..end_row {
+                            region
+                                .assign_advice(|| "assign advice rows", column, i, || Ok(F::zero()))
+                                .unwrap();
+                            region
+                                .assign_fixed(
+                                    || "assgin fixed q_usable rows ",
+                                    self.q_usable,
+                                    i,
+                                    || Ok(F::one()),
+                                )
+                                .unwrap();
```

Comment:
@DreamWuGit  you assign q_usable multi times inside each outer for loop. Better refactor the loops like this

```
for i in 0..end_row 
  assign_fixed q_usable
  for column in xxx:
    assign_advice column
     
```

Reply:
yeah, will refactor it !!

Reply:
updated it [59e83bd](https://github.com/scroll-tech/zkevm-circuits/pull/186/commits/59e83bd79bb504ea3437e88b513b31dbb4a83828) , I am thinking if all multi assigning can be regarding as errors , if yes, halo2 should check it and reports this error at the low level (a possible issue to halo2)

Reply:
@lispc  open an issue regarding this in halo2 for discussion https://github.com/zcash/halo2/issues/645